conans/client/build/cmake.py

```diff
@@ -116,7 +117,8 @@ def _generator(self):
             else:
                 return base
 
-        if self._os == "Windows":
+        # The generator depends on the build machine, not the target
+        if self._os_build == "Windows":
```

Comment:
Maybe make it a bit more backwards compatible, use ``self._os`` if self._os_build is None?

Reply:
The backward compatibility is provided by the `get_cross_building_settings` in the constructor. It gets the platform.system if the `os_build` is not defined. It "guarantees" knowing where it's running.